code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -134,17 +135,61 @@ Implant Specifics:<BR>"}
 <b>Special Features:</b> Explodes<BR>
 <b>Integrity:</b> Implant will occasionally be degraded by the body's immune system and thus will occasionally malfunction."}
 
-/obj/item/implant/dexplosive/trigger(emote, source as mob)
-	if(emote == "deathgasp")
-		src.activate("death")
+/obj/item/implant/explosive/deadman/attack_self(mob/user)
+	return
 
-/obj/item/implant/dexplosive/activate(var/cause)
-	if((!cause) || (!src.imp_in))	return 0
-	explosion(src, -1, 0, 2, 3, 0)//This might be a bit much, dono will have to see.
-	if(src.imp_in)
-		src.imp_in.gib()
+/obj/item/implant/explosive/deadman/hear(var/msg)
+	return
+
+/obj/item/implant/explosive/deadman/process()
+	if(malfunction)
+		STOP_PROCESSING(SSprocessing, src)
+		return
+	if (!implanted)
+		return
+	var/mob/M = imp_in
+
+	if(M.stat == 2)
+		activate("death")
+
+/obj/item/implant/explosive/deadman/activate(var/cause)
+	switch(cause)
+		if("death")
+			small_countdown(src)
+			STOP_PROCESSING(SSprocessing, src)
+
```

Comment:
Honestly you can do away with the switch. It only activates in one way, so just get rid of switch(cause) if("death"). And then on line 153 you can just do activate(). It won't matter that cause is null because we don't really care about the cause.